src/kbmod/search/raw_image.cpp

```diff
@@ -1,137 +1,471 @@
-            if ((p1 == NO_DATA) && (p2 != NO_DATA)) return false;
-            if ((p1 != NO_DATA) && (p2 == NO_DATA)) return false;
+    // Compute the rest of the moments.
+    for (int y = 0; y < height; ++y) {
+        for (int x = 0; x < width; ++x) {
+            int ind = y * width + x;
+            float pix_val = (pixels[ind] != NO_DATA) ? (pixels[ind] - min_val) / sum : 0.0;
 
-            // Other values match within an absolute tolerance.
-            if (fabs(p1 - p2) > atol) return false;
+            res.m00 += pix_val;
+            res.m10 += (x - c_x) * pix_val;
+            res.m20 += (x - c_x) * (x - c_x) * pix_val;
+            res.m01 += (y - c_y) * pix_val;
+            res.m02 += (y - c_y) * (y - c_y) * pix_val;
+            res.m11 += (x - c_x) * (y - c_y) * pix_val;
         }
     }
-    return true;
-}
 
-void RawImage::load_time_from_file(fitsfile* fptr) {
+    return res;
+  }
+
+
+  void RawImage::load_time_from_file(fitsfile* fptr) {
     int mjd_status = 0;
 
-    // Set object's obstime directly to -1 to indicate no time found. 
+    // Set object's obstime directly to -1 to indicate no time found.
     obstime = -1.0;
 
     // Read image observation time, trying the MJD field first and DATE-AVG second.
     // Ignore error if does not exist.
     if (fits_read_key(fptr, TDOUBLE, "MJD", &obstime, NULL, &mjd_status)) {
 
-        // Reset the status flag and try with DATE-AVG.
-        mjd_status = 0;
-        if (fits_read_key(fptr, TDOUBLE, "DATE-AVG", &obstime, NULL, &mjd_status)) {
-            obstime = -1.0;
-        }
+      // Reset the status flag and try with DATE-AVG.
+      mjd_status = 0;
+      if (fits_read_key(fptr, TDOUBLE, "DATE-AVG", &obstime, NULL, &mjd_status)) {
+        obstime = -1.0;
+      }
     }
-}
+  }
+
 
-// Load the image data from a specific layer of a FITS file.
-void RawImage::load_from_file(const std::string& file_path, int layer_num) {
+  // Load the image data from a specific layer of a FITS file.
+  void RawImage::from_fits(const std::string& file_path, int layer_num) {
     // Open the file's header and read in the obstime and the dimensions.
     fitsfile* fptr;
     int status = 0;
+    int mjdStatus = 0;
     int file_not_found;
     int nullval = 0;
     int anynull = 0;
 
     // Open the correct layer to extract the RawImage.
     std::string layerPath = file_path + "[" + std::to_string(layer_num) + "]";
     if (fits_open_file(&fptr, layerPath.c_str(), READONLY, &status)) {
-        fits_report_error(stderr, status);
-        throw std::runtime_error("Could not open FITS file to read RawImage");
+      fits_report_error(stderr, status);
+      throw std::runtime_error("Could not open FITS file to read RawImage");
     }
 
     // Read image dimensions.
     long dimensions[2];
     if (fits_read_keys_lng(fptr, "NAXIS", 1, 2, dimensions, &file_not_found, &status))
-        fits_report_error(stderr, status);
+      fits_report_error(stderr, status);
```

Comment:
I'm not sure how you got stuck in a particular commit in which this was the case, but the [1ab7772](https://github.com/dirac-institute/kbmod/pull/364/commits/1ab7772af6994cef74f8c81273a33c2cc69a9f73) was clangified and blacked before push, so this just isn't there.

I would review [4696483](https://github.com/dirac-institute/kbmod/pull/364/commits/4696483d26e944e4e120990aa08dcce9381d0a07) because it fixes the apply mask issue.